### PR TITLE
[Invoices] Adds unit level test coverage order model attributes

### PR DESCRIPTION
**Pull request by filipefurtad0**
#### What? Why?

Completes coverage for order model attributes:

```
- additional_tax_total
- included_tax_total
- shipping_method_id
- total
- number
- currency
- state
- special_instructions
```
`- payment total` (existing before this PR)
`- note` (existing before this PR)

Ran this spec locally: it passed 30/30 times.

#### What should we test?

- green build

#### Release notes

Changelog Category: Technical changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Comment by filipefurtad0** [comment]
I've tried to reproduce #11478 , and ran it 100 times locally and it always passed... So, this PR does not consider the flakyness.

**Review by dacook — COMMENTED** [review]
Hi Filipe, thanks for increasing the spec coverage! 💪 

I noticed we can simplify the order creation so have added a couple of commits, I hope they make sense. They seem to have reduced runtime from 17 -> 15 secs.

But it's very strange, the `special_instructions` seems to fail about 50% of the time on my computer. Have you any idea why?

Also I have a comment for arranging the code to consider.

**Review by mkllnk — APPROVED** [review]
This is looking good. But did you say there's a flaky spec now? I guess that needs fixing before we merge.

**Comment by dacook** [comment]
Yes, sorry I should have set it to In Dev, have done now.

**Comment by mkllnk** [comment]
> > there's a flaky spec now?
>
> I should have set it to In Dev, have done now.

Thank you for clarifying. Off topic but on the meta-level of these threads: I'm mostly including quotes like here in my replies on Github now because I read Github notifications in my email inbox and I have no idea what someone is saying "Yes" to or if they are even replying to me or someone else without these quotes. I don't want to slow you down but if that could be helpful for you, too, then we do that more often to help each other understand the context of notifications quickly.

File: spec/services/order_invoice_comparator_spec.rb [hunk 1]
```diff
@@ -26,10 +28,93 @@
         expect(subject).to be true
       end
 
-      it "returns true if a non-relevant attribute changes" do
+      it "returns true if a relevant attribute changes" do
```
[the remark this answers is not in the document]
Comment (filipefurtad0) [reply spec/services/order_invoice_comparator_spec.rb]: Thanks @dacook ,

This is a great suggestion indeed. I've made this change on commit 30e30f5a82164f088b548eeee169623a3affc2d4.

File: spec/services/order_invoice_comparator_spec.rb [hunk 2]
```diff
@@ -30,6 +30,52 @@
...
+                                      order.update_shipping_fees!
+                                    end
+        end
+
+
+        it "returns returns true if additional tax total changes" do
+
+          expect(order_with_taxes.additional_tax_total).to eq 0.11e2
+          Spree::TaxRate.first.update!(amount: 0.15)
+          order_with_taxes.create_tax_charge! && order_with_taxes.save
```
[the remark this answers is not in the document]
Comment (filipefurtad0) [reply spec/services/order_invoice_comparator_spec.rb]: Thanks for clarifying that.
Sometimes I end up sticking with the version that works, without this fine tuning. Really appreciate you take the time to write and explain these points.

I've verified the changes work (green spec), and committed here 69e1ad0ba3744e8dae21d6e12ff168ad8d2be3e1.

**Comment by dacook** [comment]
> I'm mostly including quotes like here in my replies on Github now because I read Github notifications in my email inbox and I have no idea what someone is saying "Yes" to or if they are even replying to me or someone else without these quotes. I don't want to slow you down but if that could be helpful for you, too, then we do that more often to help each other understand the context of notifications quickly.

Thanks, I find that helpful, I'll keep that in mind (but won't promise to do it every time :)
It's because of this reason that I'm in the habit of clicking "view in Github" as soon as I see anything that's not 100% clear. I often need to view it on GH anyway.

**Review by dacook — APPROVED** [review]
> No idea why @dacook . I've ran the that test case, as well as the whole spec several times, before and after the proposed changes, and have not detected any failure: it always passes on my local setup.
> 
> Not sure how to proceed, its tough when you can't spot it locally, like [this example](https://github.com/openfoodfoundation/openfoodnetwork/pull/11309#issuecomment-1701057515) as well.

Oh well, if it works consistently on  your computer and in CI then there might have been something wrong with how I was testing. 🤷

**Comment by filipefurtad0** [comment]
> I often need to view it on GH anyway

@mkllnk @dacook I find useful to add the quote indeed :-) But reading Github notifications in the inbox only can be misleading I think, especially if the message has been deleted or edited at some point.

> Oh well, if it works consistently on your computer and in CI 

Yes... I don't have a better idea. So, agree with merging till these appear in the build.

**Pull request merged**